lib/Target/TfheRustBool/TfheRustBoolEmitter.cpp

```diff
@@ -226,15 +271,18 @@ LogicalResult TfheRustBoolEmitter::printOperation(tensor::ExtractOp op) {
   return success();
 }
 
+// Need to produce a Vec<&Ciphertext>
 LogicalResult TfheRustBoolEmitter::printOperation(tensor::FromElementsOp op) {
   emitAssignPrefix(op.getResult());
   os << "vec![" << commaSeparatedValues(op.getOperands(), [&](Value value) {
     // Check if block argument, if so, clone.
-    auto cloneStr = "";
-    if (isa<BlockArgument>(value)) {
-      cloneStr = ".clone()";
-    }
-    return variableNames->getNameForValue(value) + cloneStr;
+    auto cloneStr = isa<BlockArgument>(value) ? ".clone()" : "";
+    // Get the name of defining operation its dialect
+    auto tfhe_op =
+        value.getDefiningOp()->getDialect()->getNamespace() == "tfhe_rust_bool";
```

Comment:
you could do something like this:
`llvm::isa<tfhe_rust_bool::TfheRustBoolDialect>(op.getDialect())`